src/envoy/mixer/config.cc

```diff
@@ -73,6 +79,9 @@ void ReadTransportConfig(const Json::Object& json, TransportConfig* config) {
   config->set_disable_check_cache(json.getBoolean(kDisableCheckCache, false));
   config->set_disable_quota_cache(json.getBoolean(kDisableQuotaCache, false));
   config->set_disable_report_batch(json.getBoolean(kDisableReportBatch, false));
+
+  config->set_check_cluster(json.getString(kCheckCluster, kDefaultMixerClusterName));
+  config->set_report_cluster(json.getString(kReportCluster, kDefaultMixerClusterName));
```

Comment:
This is the wrong place to set it.    Sorry for the misleading,  hhis function should be called "ReadLegacyTransportConfig"

In the mixer_filter config,  if there is a "v2" field,  
these data will be override by the data from "v2".  

https://github.com/istio/proxy/blob/master/src/envoy/mixer/config.cc#L115

You have to set these two fields after ReadV2Config() call

and check if they are not set by Pilot,  set their default value as "mixer_server"